design/one-pager-stack-relationship-labels.md

```diff
@@ -0,0 +1,229 @@
+# Stack Parent/Child Relationship Label Spec
+
+- Owner: Steven Rathbauer ([@rathpc](https://github.com/rathpc))
+- Reviewers: Crossplane Maintainers
+- Status: Draft
+
+## Proposal
+
+[crossplaneio/crossplane#752](https://github.com/crossplaneio/crossplane/issues/752)
+
+## Problem
+
+There is currently no easy way to get a list of all resources that are owned by a parent resource.
+We can leverage CRD labels to make this easier, as long as we have a standard format for doing so.
+Additionally as long as we define this label on all children CRD's, we should then be able to query
+all children of a given parent object using those labels.
+
+Per the proposal, it seems that the closest example of what I am referring to is shown on 
+[this page][common-labels] within the `app.kubernetes.io/part-of` label. However since we are not
+exactly using these labels for the intentions defined within that page, we should use something
+following those patterns instead.
+
+We had discussed using the _common labels_ as they are defined however after some deliberation we
+decided the best course of action was to make our own. _<sup>[Additional Context &rarr;](#additionalContext)</sup>_
+
+The primary reason for this was stated on [that page][common-labels]:
+
+> ### _The metadata is organized around the concept of an application._
+
+## Design
+
+> _**Important Note**: These files are post processed yaml outputs, **NOT** static package files.
+> The purpose of showing these is only to better outline and explain the relationships with respect
+> to this specific example._
+
+Wordpress example file tree structure of the example files below:
+
+```text
+stack.yaml
+|
+`-- crd.yaml
+    |
+    `-- resource.yaml
+        |-- kubernetesapplication.yaml
+        |-- kubernetescluster.yaml
+        |-- mysqlinstance.yaml
+        |-- ...
+```
+
+### Example wordpress `stack.yaml`
+
+```yaml
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: Stack
+metadata:
+  creationTimestamp: "2019-10-23T23:47:38Z"
+  generation: 1
+  name: sample-stack-wordpress
+  namespace: app-project1-dev
+  ownerReferences:
+  - apiVersion: stacks.crossplane.io/v1alpha1
+    kind: StackInstall
+    name: sample-stack-wordpress
+    uid: 86c89c94-cfc0-474e-b3b3-beffc09e7793
+  resourceVersion: "6431"
+  selfLink: /apis/stacks.crossplane.io/v1alpha1/namespaces/app-project1-dev/stacks/sample-stack-wordpress
+  uid: ec52c8c2-379f-45ec-9458-e40f070f8d2e
+  labels:
+    app.kubernetes.io/managed-by: stack-manager
+...
+```
+
+### Example wordpress `crd.yaml`
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  creationTimestamp: null
+  name: wordpressinstances.wordpress.samples.stacks.crossplane.io
+  labels:
+    core.crossplane.io/parent-kind: "stack.stacks.crossplane.io"
+    core.crossplane.io/parent-name: "sample-stack-wordpress"
```

Comment:
seems useful to also have a UID like @displague suggested in this comment:
https://github.com/crossplaneio/crossplane/pull/1091#discussion_r350961773

and @negz reinforced in https://github.com/crossplaneio/crossplane/pull/1091#discussion_r349346189

maybe `core.crossplane.io/parent-uid`?

Reply:
consider it done!